File: backend/internal/domain/publishing.go
package domain

import (
	"time"

	"github.com/google/uuid"
)

// PlatformCredential stores encrypted OAuth tokens for a platform connection.
type PlatformCredential struct {
	ID                    uuid.UUID
	WorkspaceID           uuid.UUID
	Platform              PlatformType
	ChannelID             *string
	ChannelName           *string
	EncryptedAccessToken  string
	EncryptedRefreshToken *string
	Scopes                []string
	TokenExpiresAt        *time.Time
	CreatedAt             time.Time
	UpdatedAt             time.Time
}

// ScheduledPost represents a piece of content queued for publishing to a platform.
type ScheduledPost struct {
	ID             uuid.UUID
	WorkspaceID    uuid.UUID
	ContentID      uuid.UUID
	CredentialID   *uuid.UUID
	Platform       PlatformType
	Status         string // pending, publishing, published, failed
	ScheduledAt    time.Time
	PublishedAt    *time.Time
	ErrorMessage   *string
	PlatformPostID *string
	CreatedAt      time.Time
	UpdatedAt      time.Time
}

File: frontend/packages/types/src/publishing.ts
import type { UUID, Timestamp } from './common';
import type { PlatformType } from './content';

export type ScheduledPostStatus = 'pending' | 'publishing' | 'published' | 'failed';

export interface PlatformCredential {
  id: UUID;
  workspace_id: UUID;
  platform: PlatformType;
  channel_id?: string | null;
  channel_name?: string | null;
  scopes: string[];
  token_expires_at?: Timestamp | null;
  created_at: Timestamp;
  updated_at: Timestamp;
  // Note: encrypted tokens are not exposed to the frontend
}

export interface ScheduledPost {
  id: UUID;
  workspace_id: UUID;
  content_id: UUID;
  credential_id?: UUID | null;
  platform: PlatformType;
  status: ScheduledPostStatus;
  scheduled_at: Timestamp;
  published_at?: Timestamp | null;
  error_message?: string | null;
  platform_post_id?: string | null;
  created_at: Timestamp;
  updated_at: Timestamp;
}
